sdk/src/main/java/io/radar/sdk/util/RadarLogBuffer.kt

```diff
@@ -0,0 +1,199 @@
+package io.radar.sdk.util
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import io.radar.sdk.Radar
+import io.radar.sdk.model.RadarLog
+import org.json.JSONObject
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.util.*
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.ReentrantLock
+
+/**
+ * Wraps the filesystem to support log caching
+ */
+internal class RadarLogBuffer(
```

Comment:
If we have a single threaded executor performing all operations, do we even need locking mechanisms?

I suppose we have a shared directory `RadarLogs`, but what if that wasn't the case and we auto-generated a randomized tmp folder and saved that in RadarSettings on initialize?

Reply:
right now, not everything is run on the executor - but it definitely could. Another option would be to keep the locks and use a cached thread pool. It might also be worth mentioning that Android file i/o operations already occur asynchronously, and we can tap into File Observers as well to correctly handle read/write completion events.

Reply:
I took another look at this and I think the locking mechanisms are still useful. The executor is only used for write operations (to ensure the log order).

Reply:
okay, the other thing I'm thinking of... do we need two locks? what if we just had a directory lock? would that simplify things?

Reply:
I feel it'd make it simpler to reason about, and I don't think we're going to be dealing with a lot of contention

we already have the single thread executor for writing to files, so that feels sufficient to me.

Reply:
Cleaned this up. It now only uses one lock for all these operations.